### ISSUE TITLE
Stopping Rancher Desktop kills the kim builder daemonset for good

### ISSUE THREAD

**Issue by jandubois** [rancher-sandbox/rancher-desktop#667]
When I stop and restart RD, the daemonset is scaled down to 0 instances, and will not come back up by itself:

```
kubectl get daemonset -n kube-image builder
NAME      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                          AGE
builder   0         0         0       0            0           node-role.kubernetes.io/builder=true   5m2s
```

The only way to bring it back seems to be to "Reset Kubernetes":

```
kubectl get daemonset -n kube-image builder
NAME      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                          AGE
builder   1         1         1       1            1           node-role.kubernetes.io/builder=true   27s
```

**Comment by ericpromislow** [comment]
This is at the top of `kim.log` in this situation:

```
Waiting for Kubernetes node IP to become 192.168.5.15...
Keeping builder pod kube-image:builder-mkbgf - pod IP 192.168.5.15 in 192.168.5.15,192.168.5.15
Keeping builder pod kube-image:builder-6ssxm - pod IP undefined in 192.168.5.15,192.168.5.15
Existing kim install invalid: incorrect endpoint address.
Installing kim: kim builder install --force --endpoint-addr 127.0.0.1
Error: Too many nodes, please specify a selector, e.g. kubernetes.io/hostname=lima-0
```

**Comment by mook-as** [comment]
Ah, that's probably from the rename, and should have been fixed by #662 then?

**Comment by jandubois** [comment]
> Ah, that's probably from the rename, and should have been fixed by #662 then?

That may be an issue on @ericpromislow's machine, but the problem happens even after resetting Kubernetes, at which point there should be no more references to the `lima-0` hostname.

**Comment by jandubois** [comment]
The issue turns out to be that the `node-role` label is dropped when the instance is restarted, so the daemonset kills the pod because the node selector no longer matches.

The pod can be brought back by adding the label again:

```
$ kubectl label --overwrite nodes lima-rancher-desktop node-role.kubernetes.io/builder=true
node/lima-rancher-desktop labeled
```
